tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,226 @@
+	c.T().Log("Running a scan")
+	yamlInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.RancherCisBenchmarkNamespace,
+		YAML:             fmt.Sprintf(ClusterScanYaml, scanName, c.cisProfileName),
+	}
+
+	cluster, err := client.Management.Cluster.ByID(c.project.ClusterID)
+	require.NoError(c.T(), err)
+	// Use ActionImportYaml to the apply the CIS Benchmark yaml file
+	_, err = client.Management.Cluster.ActionImportYaml(cluster, yamlInput)
+	require.NoError(c.T(), err)
+
+	steveClient, err := client.Steve.ProxyDownstream(c.project.ClusterID)
+	require.NoError(c.T(), err)
+
+	err = wait.Poll(1*time.Second, 5*time.Minute, func() (done bool, err error) {
+		// get list of constraints
+		scanList, err := steveClient.SteveType(charts.ClusterScanResourceType).List(nil)
+		if err != nil {
+			return false, err
+		}
+
+		scanIdx := -1
+
+		for idx, scan := range scanList.Data {
+			if scan.Name == scanName {
+				scanIdx = idx
+				break
+			}
+		}
+
+		if scanIdx < 0 {
+			return false, errors.New("scan not found")
+		}
+
+		clusterScanStatusType := &charts.ClusterScanStatus{}
+		scanStatus := scanList.Data[scanIdx].Status
+		err = v1.ConvertToK8sType(scanStatus, clusterScanStatusType)
+		if err != nil {
+			return false, err
+		}
+
+		if clusterScanStatusType.Display.State == scanPending ||
+			clusterScanStatusType.Display.State == scanRunning ||
+			clusterScanStatusType.Display.State == scanReporting {
+			logrus.Infof("cluster scan is %s", clusterScanStatusType.Display.State)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	require.NoError(c.T(), err)
+
+	// Call the CheckScanReport function after the scan completes
+	c.CheckScanReport(steveClient, scanName)
+}
+
+func (c *CisBenchmarkTestSuite) CheckScanReport(steveClient *v1.Client, scanName string) {
+	scanReportList, err := steveClient.SteveType(charts.ClusterScanReportType).List(nil)
+	require.NoError(c.T(), err)
+
+	scanReportIdx := -1
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
+
+	require.GreaterOrEqualf(c.T(), scanReportIdx, 0, "scan report not found")
+	cisReportSpec := &charts.ClusterScanReportSpec{}
+	scanReportSpec := scanReportList.Data[scanReportIdx].Spec
+	err = v1.ConvertToK8sType(scanReportSpec, cisReportSpec)
+	require.NoError(c.T(), err)
+
+	reportData := &charts.CisReport{}
+	err = json.Unmarshal([]byte(cisReportSpec.ReportJson), &reportData)
+	require.NoError(c.T(), err)
+	logrus.Infof("Out of total number of %d scans, %d scans passed, %d scans were skipped and %d scans failed", reportData.Total, reportData.Pass, reportData.Skip, reportData.Fail)
+	for _, group := range reportData.Results {
+		for _, check := range group.Checks {
+			if check.State == "fail" {
+				logrus.Infof("check failed: id: %s description: %s ", check.Id, check.Description)
+			} else {
+				logrus.Infof("check passed: id: %s description: %s ", check.Id, check.Description)
```

Comment:
I understand why we are printing the failed tests. Why are we printing the ones that have passed?

Reply:
Instead of logging why don't you use an assert to check these conditions. logging here doesn't make sense to me

Reply:
Updated code will list only failed tests.